### ISSUE TITLE
Can this module work with vue-class-component?

### ISSUE THREAD

**Issue by yeluoqiuzhi** [ktsn/vue-template-loader#1]
Can this module work with [vue-class-component](https://github.com/vuejs/vue-class-component)? Can you give me an example, something like best practice. I'm using typescript to write vue, .vue single file component is not friendly in ide or editor,but I need precompile template

**Comment by ktsn** [comment]
Yes, this loader just add `render` and `staticRenderFns` into Vue options object under the hood.

Example in ts:

```ts
import Vue = require('vue')
import Component from 'vue-class-component'
import withRender = require('./template.html')

@Component(withRender({
  props: ['someProp']
}))
class MyComponent extends Vue {
  someMethod () {}
}
```

Make sure to add a declaration file to let TS compiler resolve `html` import.

```ts
declare module '*.html' {
  import Vue = require('vue')
  const withRender: (options: Vue.ComponentOptions<Vue>) => Vue.ComponentOptions<Vue>
  export = withRender
}
```

BTW, I come up with to allow withRender function to use as decorator. I'm considering to implement that for next version.
That will look like following:

```ts
import Vue = require('vue')
import Component from 'vue-class-component'
import withRender = require('./template.html')

@withRender
@Component({
  props: ['someProp']
})
class MyComponent extends Vue {
  someMethod () {}
}
```

**Comment by yeluoqiuzhi** [comment]
Thansk!!! decorator in next version is so awesome.

**Comment by ktsn** [comment]
Released as [v0.2.0](https://github.com/ktsn/vue-template-loader/releases/tag/v0.2.0)
